clang/lib/Sema/SemaBounds.cpp

```diff
@@ -5978,10 +5982,22 @@ namespace {
                               CE->getNumArgs());
 
         // Concretize Call Bounds with argument expressions.
-        // We can only do this if the argument expressions are non-modifying
+        // We can only do this if the argument expressions are non-modifying.
+        // For argument expressions that are modifying, we issue an error
+        // message only in checked scope because the argument expressions may
+        // be re-evaluated during bounds validation.
+        // TODO: The long-term solution is to introduce temporaries for
+        // modifying argument expressions whose corresponding formals are used
+        // in return or parameter bounds expressions.
+        // Equality between the value computed by an argument expression and its
+        // associated temporary would also need to be recorded.
+        Sema::NonModifyingMessage Message =
```

Comment:
This seems incorrect. It can be written as:
```
Sema::NonModifyingMessage Message =
          (CSS == CheckedScopeSpecifier::CSS_Unchecked) ?
          Sema::NonModifyingMessage::NMM_None :
          Sema::NonModifyingMessage::NMM_Error;
```